lib/charms/tempo_k8s/v2/tracing.py

```diff
@@ -595,12 +656,17 @@ def publish_receivers(self, receivers: Sequence[RawReceiver]):
         for relation in self.relations:
             try:
                 TracingProviderAppData(
-                    host=self._host,
-                    external_url=self._external_url or None,
                     receivers=[
-                        Receiver(port=port, protocol=protocol) for protocol, port in receivers
+                        Receiver(
+                            url=url,
+                            protocol=ProtocolType(
+                                name=protocol,
+                                type=ReceiverProtocolType.get(protocol)
+                                or TransportProtocolType(""),
```

Comment:
what's the `TransportProtocolType("")` about? 
I'd rather do a hard `type=ReceiverProtocolType[protocol]` and let a keyerror raise.
If someone is passing an unknown protocol, we should know that sooner rather than later.